sql/scheduledlogging: TestCaptureIndexUsageStats failed

sql/scheduledlogging.TestCaptureIndexUsageStats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11769088?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11769088?buildTab=artifacts#/) on master @ [746273784cd809442ca65dc50012c58794e39cc8](https://github.com/cockroachdb/cockroach/commits/746273784cd809442ca65dc50012c58794e39cc8):


```
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719104000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719103638,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":113,"IndexID":2,"DatabaseName":"test2","TableName":"test2_table","IndexName":"test2_table_letter_idx","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.580922 +0000 UTC","SchemaName":"public"} nsql1 16 true TELEMETRY 326 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719086000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719085452,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":113,"IndexID":1,"DatabaseName":"test2","TableName":"test2_table","IndexName":"test2_table_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.108553 +0000 UTC","SchemaName":"public"} nsql1 15 true TELEMETRY 334 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719068000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719067745,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":112,"IndexID":2,"DatabaseName":"test","TableName":"test_table","IndexName":"test_table_letter_idx","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.137695 +0000 UTC","SchemaName":"public"} nsql1 14 true TELEMETRY 323 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719053000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719052449,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":112,"IndexID":1,"DatabaseName":"test","TableName":"test_table","IndexName":"test_table_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.094073 +0000 UTC","SchemaName":"public"} nsql1 13 true TELEMETRY 331 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719037000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719036489,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":114,"IndexID":2,"DatabaseName":"\"mIxEd-CaSe\"\"woo☃\"","TableName":"sPe-CiAl✔","IndexName":"IdX✏","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.886968 +0000 UTC","SchemaName":"public"} nsql1 12 true TELEMETRY 329 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719014000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719013028,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":114,"IndexID":1,"DatabaseName":"\"mIxEd-CaSe\"\"woo☃\"","TableName":"sPe-CiAl✔","IndexName":"sPe-CiAl✔_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.118176 +0000 UTC","SchemaName":"public"} nsql1 11 true TELEMETRY 353 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600718981000 4732 util/log/event_log.go 32 {"Timestamp":1694649600718980537,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":115,"IndexID":2,"DatabaseName":"index","TableName":"index","IndexName":"index","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:53.217158 +0000 UTC","SchemaName":"public"} nsql1 10 true TELEMETRY 303 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600718934000 4732 util/log/event_log.go 32 {"Timestamp":1694649600718930174,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":115,"IndexID":1,"DatabaseName":"index","TableName":"index","IndexName":"index_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.127195 +0000 UTC","SchemaName":"public"} nsql1 9 true TELEMETRY 322 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589118000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589117604,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":113,"IndexID":2,"DatabaseName":"test2","TableName":"test2_table","IndexName":"test2_table_letter_idx","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.580922 +0000 UTC","SchemaName":"public"} nsql1 8 true TELEMETRY 326 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589110000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589110571,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":113,"IndexID":1,"DatabaseName":"test2","TableName":"test2_table","IndexName":"test2_table_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.108553 +0000 UTC","SchemaName":"public"} nsql1 7 true TELEMETRY 334 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589103000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589103581,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":112,"IndexID":2,"DatabaseName":"test","TableName":"test_table","IndexName":"test_table_letter_idx","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.137695 +0000 UTC","SchemaName":"public"} nsql1 6 true TELEMETRY 323 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589098000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589097865,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":112,"IndexID":1,"DatabaseName":"test","TableName":"test_table","IndexName":"test_table_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.094073 +0000 UTC","SchemaName":"public"} nsql1 5 true TELEMETRY 331 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589092000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589091563,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":114,"IndexID":2,"DatabaseName":"\"mIxEd-CaSe\"\"woo☃\"","TableName":"sPe-CiAl✔","IndexName":"IdX✏","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.886968 +0000 UTC","SchemaName":"public"} nsql1 4 true TELEMETRY 329 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589083000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589082880,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":114,"IndexID":1,"DatabaseName":"\"mIxEd-CaSe\"\"woo☃\"","TableName":"sPe-CiAl✔","IndexName":"sPe-CiAl✔_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.118176 +0000 UTC","SchemaName":"public"} nsql1 3 true TELEMETRY 353 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589059000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589053158,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":115,"IndexID":2,"DatabaseName":"index","TableName":"index","IndexName":"index","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:53.217158 +0000 UTC","SchemaName":"public"} nsql1 2 true TELEMETRY 303 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649595661827000 4732 util/log/event_log.go 32 {"Timestamp":1694649595661822893,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":115,"IndexID":1,"DatabaseName":"index","TableName":"index","IndexName":"index_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.127195 +0000 UTC","SchemaName":"public"} nsql1 1 true TELEMETRY 322 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:342: found index counts: map[IdX✏:2 index:2 index_pkey:2 sPe-CiAl✔_pkey:2 test2_table_letter_idx:2 test2_table_pkey:2 test_table_letter_idx:2 test_table_pkey:2]
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649604447272000 4732 util/log/event_log.go 32 {"Timestamp":1694649604447271735,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":113,"IndexID":2,"DatabaseName":"test2","TableName":"test2_table","IndexName":"test2_table_letter_idx","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.580922 +0000 UTC","SchemaName":"public"} nsql1 24 true TELEMETRY 326 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649604447256000 4732 util/log/event_log.go 32 {"Timestamp":1694649604447255897,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":113,"IndexID":1,"DatabaseName":"test2","TableName":"test2_table","IndexName":"test2_table_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.108553 +0000 UTC","SchemaName":"public"} nsql1 23 true TELEMETRY 334 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649604447239000 4732 util/log/event_log.go 32 {"Timestamp":1694649604447239109,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":112,"IndexID":2,"DatabaseName":"test","TableName":"test_table","IndexName":"test_table_letter_idx","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.137695 +0000 UTC","SchemaName":"public"} nsql1 22 true TELEMETRY 323 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649604447227000 4732 util/log/event_log.go 32 {"Timestamp":1694649604447226536,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":112,"IndexID":1,"DatabaseName":"test","TableName":"test_table","IndexName":"test_table_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.094073 +0000 UTC","SchemaName":"public"} nsql1 21 true TELEMETRY 331 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649604447214000 4732 util/log/event_log.go 32 {"Timestamp":1694649604447213731,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":114,"IndexID":2,"DatabaseName":"\"mIxEd-CaSe\"\"woo☃\"","TableName":"sPe-CiAl✔","IndexName":"IdX✏","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.886968 +0000 UTC","SchemaName":"public"} nsql1 20 true TELEMETRY 329 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649604447199000 4732 util/log/event_log.go 32 {"Timestamp":1694649604447197573,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":114,"IndexID":1,"DatabaseName":"\"mIxEd-CaSe\"\"woo☃\"","TableName":"sPe-CiAl✔","IndexName":"sPe-CiAl✔_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.118176 +0000 UTC","SchemaName":"public"} nsql1 19 true TELEMETRY 353 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649604447181000 4732 util/log/event_log.go 32 {"Timestamp":1694649604447180601,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":115,"IndexID":2,"DatabaseName":"index","TableName":"index","IndexName":"index","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:53.217158 +0000 UTC","SchemaName":"public"} nsql1 18 true TELEMETRY 303 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649604447143000 4732 util/log/event_log.go 32 {"Timestamp":1694649604447138889,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":115,"IndexID":1,"DatabaseName":"index","TableName":"index","IndexName":"index_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.127195 +0000 UTC","SchemaName":"public"} nsql1 17 true TELEMETRY 322 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719104000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719103638,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":113,"IndexID":2,"DatabaseName":"test2","TableName":"test2_table","IndexName":"test2_table_letter_idx","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.580922 +0000 UTC","SchemaName":"public"} nsql1 16 true TELEMETRY 326 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719086000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719085452,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":113,"IndexID":1,"DatabaseName":"test2","TableName":"test2_table","IndexName":"test2_table_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.108553 +0000 UTC","SchemaName":"public"} nsql1 15 true TELEMETRY 334 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719068000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719067745,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":112,"IndexID":2,"DatabaseName":"test","TableName":"test_table","IndexName":"test_table_letter_idx","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.137695 +0000 UTC","SchemaName":"public"} nsql1 14 true TELEMETRY 323 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719053000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719052449,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":112,"IndexID":1,"DatabaseName":"test","TableName":"test_table","IndexName":"test_table_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.094073 +0000 UTC","SchemaName":"public"} nsql1 13 true TELEMETRY 331 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719037000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719036489,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":114,"IndexID":2,"DatabaseName":"\"mIxEd-CaSe\"\"woo☃\"","TableName":"sPe-CiAl✔","IndexName":"IdX✏","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.886968 +0000 UTC","SchemaName":"public"} nsql1 12 true TELEMETRY 329 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600719014000 4732 util/log/event_log.go 32 {"Timestamp":1694649600719013028,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":114,"IndexID":1,"DatabaseName":"\"mIxEd-CaSe\"\"woo☃\"","TableName":"sPe-CiAl✔","IndexName":"sPe-CiAl✔_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.118176 +0000 UTC","SchemaName":"public"} nsql1 11 true TELEMETRY 353 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600718981000 4732 util/log/event_log.go 32 {"Timestamp":1694649600718980537,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":115,"IndexID":2,"DatabaseName":"index","TableName":"index","IndexName":"index","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:53.217158 +0000 UTC","SchemaName":"public"} nsql1 10 true TELEMETRY 303 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649600718934000 4732 util/log/event_log.go 32 {"Timestamp":1694649600718930174,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":115,"IndexID":1,"DatabaseName":"index","TableName":"index","IndexName":"index_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.127195 +0000 UTC","SchemaName":"public"} nsql1 9 true TELEMETRY 322 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589118000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589117604,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":113,"IndexID":2,"DatabaseName":"test2","TableName":"test2_table","IndexName":"test2_table_letter_idx","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.580922 +0000 UTC","SchemaName":"public"} nsql1 8 true TELEMETRY 326 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589110000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589110571,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":113,"IndexID":1,"DatabaseName":"test2","TableName":"test2_table","IndexName":"test2_table_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.108553 +0000 UTC","SchemaName":"public"} nsql1 7 true TELEMETRY 334 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589103000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589103581,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":112,"IndexID":2,"DatabaseName":"test","TableName":"test_table","IndexName":"test_table_letter_idx","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.137695 +0000 UTC","SchemaName":"public"} nsql1 6 true TELEMETRY 323 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589098000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589097865,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":112,"IndexID":1,"DatabaseName":"test","TableName":"test_table","IndexName":"test_table_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.094073 +0000 UTC","SchemaName":"public"} nsql1 5 true TELEMETRY 331 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589092000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589091563,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":114,"IndexID":2,"DatabaseName":"\"mIxEd-CaSe\"\"woo☃\"","TableName":"sPe-CiAl✔","IndexName":"IdX✏","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:52.886968 +0000 UTC","SchemaName":"public"} nsql1 4 true TELEMETRY 329 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589083000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589082880,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":114,"IndexID":1,"DatabaseName":"\"mIxEd-CaSe\"\"woo☃\"","TableName":"sPe-CiAl✔","IndexName":"sPe-CiAl✔_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.118176 +0000 UTC","SchemaName":"public"} nsql1 3 true TELEMETRY 353 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649596589059000 4732 util/log/event_log.go 32 {"Timestamp":1694649596589053158,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":115,"IndexID":2,"DatabaseName":"index","TableName":"index","IndexName":"index","IndexType":"secondary","CreatedAt":"2023-09-13 23:59:53.217158 +0000 UTC","SchemaName":"public"} nsql1 2 true TELEMETRY 303 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:326: checking entry: {INFO 1694649595661827000 4732 util/log/event_log.go 32 {"Timestamp":1694649595661822893,"EventType":"captured_index_usage_stats","LastRead":"0001-01-01 00:00:00 +0000 UTC","TableID":115,"IndexID":1,"DatabaseName":"index","TableName":"index","IndexName":"index_pkey","IndexType":"primary","IsUnique":true,"CreatedAt":"2023-09-13 23:59:52.127195 +0000 UTC","SchemaName":"public"} nsql1 1 true TELEMETRY 322 0 0 10 tenant-10}
    captured_index_usage_stats_test.go:342: found index counts: map[IdX✏:3 index:3 index_pkey:3 sPe-CiAl✔_pkey:3 test2_table_letter_idx:3 test2_table_pkey:3 test_table_letter_idx:3 test_table_pkey:3]
    captured_index_usage_stats_test.go:288: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/scheduledlogging/captured_index_usage_stats_test.go:288
        	Error:      	"1s" is not greater than "1.057107s"
        	Test:       	TestCaptureIndexUsageStats
        	Messages:   	1s <= 5.057107s-4s
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/69cc8b5ff8b35dcd9569961cf48105ef/logTestCaptureIndexUsageStats354847578
--- FAIL: TestCaptureIndexUsageStats (17.02s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cluster-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCaptureIndexUsageStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31526

--- a/pkg/sql/scheduledlogging/captured_index_usage_stats_test.go
+++ b/pkg/sql/scheduledlogging/captured_index_usage_stats_test.go
@@ -78,9 +78,6 @@ func TestCaptureIndexUsageStats(t *testing.T) {
 	stubScheduleCheckEnabledInterval := time.Second
 	stubLoggingDelay := 0 * time.Second
 
-	// timeBuffer is a short time buffer to account for non-determinism in the logging timings.
-	const timeBuffer = 4 * time.Second
-
 	schedulesFinishedChan := make(chan struct{})
 	logsVerifiedChan := make(chan struct{})
 
@@ -90,7 +87,7 @@ func TestCaptureIndexUsageStats(t *testing.T) {
 				getLoggingDuration: sd.getLoggingDuration,
 				getOverlapDuration: sd.getOverlapDuration,
 				onScheduleComplete: func() {
-					// Notify that the scheudle has completed.
+					// Notify that the schedule has completed.
 					schedulesFinishedChan <- struct{}{}
 					// Wait for logs to be verified before proceeding.
 					<-logsVerifiedChan
@@ -174,10 +171,8 @@ func TestCaptureIndexUsageStats(t *testing.T) {
 
 	// Enable capture of index usage stats.
 	telemetryCaptureIndexUsageStatsEnabled.Override(context.Background(), &tenantSettings.SV, true)
-	//t.Fatal(telemetryCaptureIndexUsageStatsEnabled.Get(&s.ClusterSettings().SV))
 
 	// Check that telemetry log file contains all the entries we're expecting, at the scheduled intervals.
-
 	expectedTotalNumEntriesInSingleInterval := 8
 	expectedNumberOfIndividualIndexEntriesInSingleInterval := 1
 
@@ -257,38 +252,6 @@ func TestCaptureIndexUsageStats(t *testing.T) {
 	sort.Slice(entries, func(a int, b int) bool {
 		return entries[a].Time < entries[b].Time
 	})
-
-	testData := []time.Duration{
-		0 * time.Second,
-		// the difference in number of seconds between first and second schedule
-		stubScheduleInterval - firstScheduleLoggingDuration,
-		// the difference in number of seconds between second and third schedule
-		sd.getOverlapDuration(),
-	}
-
-	var (
-		previousTimestamp = int64(0)
-		currentTimestamp  = int64(0)
-	)
-
-	// Check the timestamp differences between schedules.
-	for idx, expectedDuration := range testData {
-		entriesLowerBound := idx * expectedTotalNumEntriesInSingleInterval
-		entriesUpperBound := (idx + 1) * expectedTotalNumEntriesInSingleInterval
-		scheduleEntryBlock := entries[entriesLowerBound:entriesUpperBound]
-		// Take the first log entry from the schedule.
-		currentTimestamp = scheduleEntryBlock[0].Time
-		// If this is the first iteration, initialize the previous timestamp.
-		if idx == 0 {
-			previousTimestamp = currentTimestamp
-		}
-
-		actualDuration := time.Duration(currentTimestamp - previousTimestamp)
-		// Use a time window to afford some non-determinism in the test.
-		require.Greaterf(t, expectedDuration, actualDuration-timeBuffer, "%v <= %v-%v", expectedDuration, actualDuration, timeBuffer)
-		require.Greater(t, actualDuration+timeBuffer, expectedDuration, "%v+%v <= %v", expectedDuration, actualDuration, timeBuffer)
-		previousTimestamp = currentTimestamp
-	}
 
 }
 

